### ISSUE TITLE
Console 404/Mixed Content Error on Checkout

### ISSUE THREAD

**Issue by ericboles** [woothemes/woocommerce#7683]
Been seeing a console error on submissions on the checkout page during the AJAX calls to check the checkout form on WC 2.3.  A call seemingly at the end of that process is trying to retrieve content from http://www.store.com/checkout/undefined. There's obviously nothing at that URL and a 404 error is subsequently shown in the console. Orders are able to complete and go through despite the error, but some browsers (IE) are throwing a mixed content error which can be confusing for customers.

WooCommerce settings are correct and the pages are correctly assigned. Plugins have been deactivated and the issue still remains on a clean Twenty Fifteen theme. This install is using the latest WooCommerce + WC Subscriptions + Stripe.

Screenshot is attached showing details from console error in Chrome:
![checkout_console](https://cloud.githubusercontent.com/assets/6694920/6567690/7cf801d2-c691-11e4-8456-2a28b01771f6.jpg)
.

**Comment by slash1andy** [comment]
Hey there,

I wasn't able to replicate this (trying at http://demandbrew.com since there wasn't a URL in the actual issue). The console looked really clean on checkout (went up until the point to click to submit my order).

You are probably better off to submit a ticket here: https://woothemes.zendesk.com

**Comment by ericboles** [comment]
Thanks, Andy.  The issue crops up once the form is submitted--seemingly during the AJAX check of the form. The console is clear before that step.

I've setup a test site that's about as clean as can get--the only plugins are WooCommerce, WooCommerce Subscriptions and WooCommerce Stripe and the Storefront Theme.  Stripe is setup with test keys so you can see the issue without needing a real credit card.

http://testingbrew.wpengine.com/

I believe an actual checkout completes goes quickly to see the 404 appear in the console, but entering an invalid credit card number (e.g. 4242 4242 4242 4241) and then submitting the form highlights the issue without moving beyond the checkout page. (You don't even need the address info in the form).

I'd appreciate if you could look into this further.  I can submit a ticket at WooThemes, but I'd like to confirm that it isn't a WooCommerce issue before then.

Thanks again.

**Comment by mikejolley** [comment]
Impossible to see whats causing this from the outside - I don't think its core or we'd all be seeing it. This needs more investigation on support.woo where we can get access to admin.

**Comment by ericboles** [comment]
Mike--  From further investigation, this isn't in core.  It seems to be an issue with an AJAX call in stripe.js in the Stripe extension. I've just submitted a ticket with WooThemes, but since you're the developer there, might as well flag it here for you.

From some investigation, the culprit seems to be line 41 in stripe.js--updating it to match the code from a similar call in simplify commerce fixes the issue: 

Line 41 from stripe.js:
```javascript
$form.block({message: null, overlayCSS: {background: '#fff url(' + woocommerce_params.ajax_loader_url + ') no-repeat center', opacity: 0.6}});
```
Replaced with lines 12-18 from /woocommerce/gateways/simplify-commerce/assets/js/simplify-commerce.js:
```javascript
$form.block({
					message: null,
					overlayCSS: {
						background: '#fff',
						opacity: 0.6
					}
				});
```

Looks like the Simplify Commerce JS was recently updated for WC 2.3 to remove the ajax_loader_url reference, but the reference still exists for Stripe.  Maybe some deprecation from WC 2.3?
